File: backend/src/routes/notes.routes.ts
import { Router } from 'express'
import { requireAuth } from '../middlewares/auth.middleware'
import { listarNotas, criarNota, editarNota, deletarNota, togglePin } from '../controllers/notes.controller'

const router = Router()

// Todas as rotas de notas exigem autenticação
router.use(requireAuth)

// GET  /api/notes     → lista todas as notas do usuário
// POST /api/notes     → cria nova nota
router.get('/',  listarNotas)
router.post('/', criarNota)

// PUT    /api/notes/:id      → edita campos da nota
// DELETE /api/notes/:id      → remove a nota
router.put('/:id',    editarNota)
router.delete('/:id', deletarNota)

// PATCH /api/notes/:id/pin   → alterna pin  { isPinned: boolean }
router.patch('/:id/pin', togglePin)

export default router

File: backend/internal/notes/notes_routes.go
package notes

import (
	"github.com/gin-gonic/gin"
	"github.com/jackc/pgx/v5/pgxpool"

	"beeplanerofc/backend-go/internal/middleware"
	"beeplanerofc/backend-go/internal/supabaseauth"
)

// RegisterRoutes registra as rotas de notas no grupo /api/notes. Todas exigem login.
func RegisterRoutes(router *gin.RouterGroup, db *pgxpool.Pool, authClient *supabaseauth.Client) {
	ctrl := NewController(db)

	router.Use(middleware.RequireAuth(db, authClient))

	router.GET("", ctrl.ListarNotas)
	router.POST("", ctrl.CriarNota)
	router.PUT("/:id", ctrl.EditarNota)
	router.DELETE("/:id", ctrl.DeletarNota)
	router.PATCH("/:id/pin", ctrl.TogglePin)
}
